thirdparty/icu4c/meson.build

```diff
@@ -0,0 +1,255 @@
+if get_option('builtin_icu')
+    _builtin_icu_incdirs = include_directories('common', 'i18n')
+
+    _builtin_icu_srcs = files(
```

Comment:
it's inconsistent - for Godot source code, you decide to use auto-discovery files, but for third-party explicit list. 
It is much better to use explicit everywhere.